test/jdk/javax/swing/plaf/basic/BasicTreeUI/TreeCellRendererLeakTest.java

```diff
@@ -0,0 +1,206 @@
+    private JTabbedPane jTabbedPane1;
+    private JTree jTree1;
+
+    // JLabel with an instance counter
+    public class TestLabel extends JLabel {
+        public TestLabel() {
+            smCount++;
+        }
+
+        public void finalize( ) {
+            smCount--;
+        }
+    }
+
+    // Custom TreeCellRenderer
+    public class TreeCellRenderer extends DefaultTreeCellRenderer {
+
+        public TreeCellRenderer( ) {
+        }
+
+        // Create a new JLabel every time
+        public Component getTreeCellRendererComponent(
+                JTree tree,
+                Object value,
+                boolean sel,
+                boolean expanded,
+                boolean leaf,
+                int row,
+                boolean hasFocus) {
+            JLabel label = new TestLabel( );
+            label.setText("TreeNode: " + value.toString());
+            if (sel) {
+                label.setBackground(getBackgroundSelectionColor());
+            } else {
+                label.setBackground(getBackgroundNonSelectionColor());
+            }
+
+            return label;
+        }
+    }
+
+    public TreeCellRendererLeakTest() {
+        initComponents();
+        jTree1.setCellRenderer(new TreeCellRenderer());
+        Thread updateThread = new Thread(new Runnable( ) {
+            public void run( ) {
+                runChanges( );
+            }
+        });
+        updateThread.setDaemon(true);
+        updateThread.start();
+        Thread infoThread = new Thread(new Runnable( ) {
+            public void run( ) {
+                runInfo( );
+            }
+        });
+        infoThread.setDaemon(true);
+        infoThread.start();
+    }
+
+    // <editor-fold defaultstate="collapsed" desc=" Generated Code ">
+    private void initComponents() {
+        jTabbedPane1 = new javax.swing.JTabbedPane();
+        jPanel1 = new javax.swing.JPanel();
+        jScrollPane1 = new javax.swing.JScrollPane();
+        jTree1 = new javax.swing.JTree();
+        jPanel2 = new javax.swing.JPanel();
+
+        jPanel1.setLayout(new java.awt.BorderLayout());
+
+        jScrollPane1.setViewportView(jTree1);
+
+        jPanel1.add(jScrollPane1, java.awt.BorderLayout.CENTER);
+
+        jTabbedPane1.addTab("tab1", jPanel1);
+
+        jPanel2.setLayout(new java.awt.BorderLayout());
+
+        jTabbedPane1.addTab("tab2", jPanel2);
+
+        jTabbedPane1.setSelectedIndex(1);
+
+        frame = new JFrame();
+        frame.getContentPane().add(jTabbedPane1, java.awt.BorderLayout.CENTER);
+
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }// </editor-fold>
+
+    public static void main(String args[]) throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(() -> {
+                TreeCellRendererLeakTest tf = new TreeCellRendererLeakTest();
+            });
+            while (!done) {
+                Thread.sleep(100);
+            }
```

Comment:
Can't we use [`CountDownLatch`](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/CountDownLatch.html) instead of a boolean flag and busy-waiting?